fedimint-client/src/sm/executor.rs

```diff
@@ -1076,11 +1202,14 @@ mod tests {
                 broadcast: broadcast.clone(),
             },
         );
-        let executor = executor_builder
+        let (executor, sm_update_rx) = executor_builder
             .build(db.clone(), Notifier::new(db.clone()))
             .await;
         executor
-            .start_executor(Arc::new(|_, _| DynGlobalClientContext::new_fake()))
+            .start_executor(
+                Arc::new(|_, _| DynGlobalClientContext::new_fake()),
+                sm_update_rx,
+            )
```

Comment:
If we just return `sm_update_rx` here to give it back, why not put that `Mutex<Option<…>>` into `Executor` to avoid this?

Reply:
Felt like unnecessary mutability. :D

But I guess it can be done.

Reply:
Didn't you do exactly this on the client level instead? Will give the PR a complete review again, then I'll see if it was about the same oject.